aws/validators.go

```diff
@@ -117,15 +117,10 @@ func validateTransferServerID(v interface{}, k string) (ws []string, errors []er
 
 func validateTransferUserName(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Nit: In the future this can be simplified to use Terraform Provider SDK validation functions to show multiple validation errors easier:

```go
ValidateFunc: validation.All(
	validation.StringLenBetween(3, 32),
	validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_]`), "must begin with alphanumeric or underscore character"),
	validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9-_]+$`), "must contain only alphanumeric, hyphen, and underscore characters"),
),
```

Reply:
Thanks, I wasn't aware of this. Is there a tech-debt effort to change other validation functions as well in `validators.go`? I'd be happy to help with that.

Reply:
I thought I had created the technical debt issue for that, but apparently not. I'll try to get that created later today.

For now, this is a good start on a similar effort: #7906

Reply:
Here's an initial proposal issue for fixing up `validators.go`: #8424 We should wait until other maintainers submit input to it before beginning any implementation.